### PR TITLE
✨ JS crash reporter

### PR DESCRIPTION
**Pull request by salmatarzi**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -81,6 +84,15 @@ module.exports = {
             Instabug.setUserStepsEnabled(isUserStepsEnabled);
     },
 
+    /**
+     * Report un-caught exceptions to Instabug dashboard
+     * We don't send exceptions from __DEV__, since it's way too noisy!
+     */
+    setCrashReportingEnabled: function(enableCrashReporter){
+      if (Platform.OS === 'ios')
+        Instabug.setCrashReportingEnabled(enableCrashReporter);
```
Comment (DevHossamHassan) [inline index.js]: @salmatarzi Why are we enabling this API for ios only?

File: android/build.gradle [hunk 2]
```diff
@@ -21,6 +21,6 @@ android {
 
 dependencies {
     compile 'com.facebook.react:react-native:0.20.+'
-    compile 'com.instabug.library:instabug:4.9.1'
+    compile 'com.instabug.library:instabug:4.5.0.17-SNAPSHOT''
```
Comment (DevHossamHassan) [inline android/build.gradle]: @salmatarzi Why older version?

File: android/src/main/java/com/instabug/reactlibrary/RNInstabugReactnativeModule.java [hunk 3]
```diff
@@ -378,7 +464,7 @@ public void setEnabledAttachmentTypes(boolean screenshot, boolean extraScreensho
             galleryImage, boolean screenRecording) {
         try {
             Instabug.setAttachmentTypesEnabled(screenshot, extraScreenshot, galleryImage,
-                    screenRecording);
+                    false, screenRecording);
```
Comment (DevHossamHassan) [inline android/src/main/java/com/instabug/reactlibrary/RNInstabugReactnativeModule.java]: Why are we adding false here?

File: android/src/main/java/com/instabug/reactlibrary/RNInstabugReactnativeModule.java [hunk 4]
```diff
@@ -292,6 +302,82 @@ public void setFileAttachment(String fileUri, String fileNameWithExtension) {
...
+     *
+     * @param isEnabled   Exception object to be sent to Instabug's servers
+     */
+    @ReactMethod
+    public void setCrashReportingEnabled(boolean isEnabled) {
+        try {
+            if(isEnabled) {
+                Instabug.setCrashReportingState(Feature.State.ENABLED);
+            } else {
+
```
Comment (DevHossamHassan) [inline android/src/main/java/com/instabug/reactlibrary/RNInstabugReactnativeModule.java]: please remove this redundant line

File: ios/InstabugCore.framework/Headers/IBGTypes.h [hunk 5]
```diff
@@ -5,7 +5,7 @@
  
  Copyright:  (c) 2013-2018 by Instabug, Inc., all rights reserved.
  
- Version:    7.8.3
+ Version:    0.0.0
```
Comment (yousefhamza) [inline ios/InstabugCore.framework/Headers/IBGTypes.h]: Version string baz kda leh 🤔

File: ios/Instabug.framework/Headers/Instabug.h [hunk 6]
```diff
@@ -5,7 +5,7 @@
 
  Copyright:  (c) 2013-2018 by Instabug, Inc., all rights reserved.
 
- Version:    7.8.3
+ Version:    0.0.0
```
Comment (yousefhamza) [inline ios/Instabug.framework/Headers/Instabug.h]: Also here

File: ios/RNInstabug/InstabugReactBridge.m [hunk 7]
```diff
@@ -58,6 +61,20 @@ - (dispatch_queue_t)methodQueue {
     [Instabug setFileAttachment:fileLocation];
 }
 
+RCT_EXPORT_METHOD(sendJSCrash:(NSDictionary *)stackTrace) {
+    SEL reportCrashWithStackTraceSEL = NSSelectorFromString(@"reportCrashWithStackTrace:handled:");
+    if ([[Instabug class] respondsToSelector:reportCrashWithStackTraceSEL]) {
```
Comment (yousefhamza) [inline ios/RNInstabug/InstabugReactBridge.m]: I think `class` isn't needed here

File: ios/RNInstabug/InstabugReactBridge.m [hunk 8]
```diff
@@ -34,8 +34,11 @@ - (dispatch_queue_t)methodQueue {
     [Instabug startWithToken:token invocationEvent:invocationEvent];
     RCTAddLogFunction(InstabugReactLogFunction);
     RCTSetLogThreshold(RCTLogLevelInfo);
-    [Instabug setCrashReportingEnabled:NO];
     [Instabug setNetworkLoggingEnabled:NO];
+    SEL setCrossPlatformSEL = NSSelectorFromString(@"setCrossPlatform:");
+    if ([[Instabug class] respondsToSelector:setCrossPlatformSEL]) {
+        [[Instabug class] performSelector:setCrossPlatformSEL withObject:@(true)];
```
Comment (yousefhamza) [inline ios/RNInstabug/InstabugReactBridge.m]: `@(YES)` instead of `@(true)`

Comment (yousefhamza) [reply ios/RNInstabug/InstabugReactBridge.m]: ### Suggestion for future progress on this:
Take a look at `NSInvocation`, I think it's better for handling calls with parameters (like more than 1 which `performSelector` doesn't support)

**Documentation:**
https://developer.apple.com/documentation/foundation/nsinvocation

**Example:**
```
NSInvocation *inv = [NSInvocation invocationWithMethodSignature:[self.tableView methodSignatureForSelector:@selector(setEditing:Animated:)]];
[inv setSelector:@selector(setEditing:Animated:)];
[inv setTarget:self.tableView];
[inv setArgument:&yes atIndex:2]; //this is the editing BOOL (0 and 1 are explained in the link above)
[inv setArgument:&yes atIndex:3]; //this is the animated BOOL
[inv invoke]
```

**Comment by chadyj** [comment]
Any progress on bringing crash reporting to RN?

**Comment by salmatarzi** [comment]
@chadyj Yes, it's expected to be out on the 15th of April

**Comment by chadyj** [comment]
Did this make it into https://github.com/Instabug/instabug-reactnative/releases/tag/v2.9.0
?

**Comment by salmatarzi** [comment]
@Jonovono @chadyj This has been released at last ✨🎉

**Comment by chadyj** [comment]
Yay thank you! 🍻

**Pull request merged**